fendermint/vm/interpreter/src/chain.rs

```diff
@@ -34,6 +34,40 @@ impl<I> ChainMessageInterpreter<I> {
     }
 }
 
+#[async_trait]
+impl<I> ProposalInterpreter for ChainMessageInterpreter<I>
+where
+    I: Sync + Send,
+{
+    // TODO: The state can include the IPLD Resolver mempool, for example by using STM
+    // to implement a shared memory space.
+    type State = ();
+    type Message = ChainMessage;
+
+    /// Check whether there are any "ready" messages in the IPLD resolution mempool which can be appended to the proposal.
+    ///
+    /// We could also use this to select the most profitable user transactions, within the gas limit. We can also take into
+    /// account the transactions which are part of top-down or bottom-up checkpoints, to stay within gas limits.
+    async fn prepare(
+        &self,
+        _state: Self::State,
+        msgs: Vec<Self::Message>,
+    ) -> anyhow::Result<Vec<Self::Message>> {
+        // For now this is just a placeholder.
+        Ok(msgs)
+    }
+
+    /// Perform finality checks on top-down transactions and availability checks on bottom-up transactions.
+    async fn process(
```

Comment:
maybe we can call it `validate` or something else along the line as process I feel it's consuming or processing the message, but we are actually inspecting.

Reply:
Yeah I know what you mean, I based all names in all the interfaces on what they are called in ABCI but without their prefix they are a bit confusing, like `begin` and `end`, or this `process` - they make sense in the context of their trait, but when the interpreter implements all interfaces, they are not very good. 

Should we just use the full names, like `begin_block` and `process_proposal`?

Reply:
`validate` would be similar to `process` in that it would be clashing with `check`, which lives in the `CheckInterpreter`.

Reply:
Created https://github.com/consensus-shipyard/fendermint/issues/207